### ISSUE TITLE
AttributeError: 'str' object has no attribute 'pre'

### ISSUE THREAD

**Issue by mcandre** [pyinvoke/invoke#160]
I want to lint my Python code, using invoke to call the individual lint programs. Instead of running the programs, invoke throws an error.

Source:

https://github.com/mcandre/version/tree/master/python

Trace:

```
$ invoke lint
Traceback (most recent call last):
  File "/usr/local/bin/invoke", line 9, in <module>
    load_entry_point('invoke==0.8.2', 'console_scripts', 'invoke')()
  File "/usr/local/lib/python2.7/dist-packages/invoke/cli.py", line 295, in main
    dispatch(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/invoke/cli.py", line 288, in dispatch
    return executor.execute(*tasks, dedupe=dedupe)
  File "/usr/local/lib/python2.7/dist-packages/invoke/executor.py", line 76, in execute
    tasks = self._dedupe(self._expand_tasks(tasks), dedupe)
  File "/usr/local/lib/python2.7/dist-packages/invoke/executor.py", line 133, in _expand_tasks
    ret.extend(self._expand_tasks(task.pre))
  File "/usr/local/lib/python2.7/dist-packages/invoke/executor.py", line 133, in _expand_tasks
    ret.extend(self._expand_tasks(task.pre))
AttributeError: 'str' object has no attribute 'pre'
```

System:

```
$ specs invoke python os
Specs:

specs 0.8
https://github.com/mcandre/specs#readme

invoke --version
Invoke 0.8.2

pip --version
pip 1.5.4 from /usr/lib/python2.7/dist-packages (python 2.7)

python --version
Python 2.7.6

lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04 LTS
Release:    14.04
Codename:   trusty
```

**Comment by mcandre** [comment]
Is this issue fixed in a release since 0.8.2?

**Comment by bitprophet** [comment]
It's not an issue, it is a backwards compatible change, that is why I was telling you to look at the changelog :) See the second big red box here: http://www.pyinvoke.org/changelog.html#0.8.0

tl;dr change `pre=["my task"]` to `pre=[mytask]` and it ought to work better.

**Comment by mcandre** [comment]
Thanks for pointing that out. This works for me, with invoke 0.10!

```
from invoke import run, task

@task
def pep8():
  run("pep8 .")

@task
def pylint():
  run("pylint *.py")

@task
def pyflakes():
  run("pyflakes .")

@task(pre=[pep8, pylint, pyflakes])
def lint():
  pass
```

**Comment by bitprophet** [comment]
Exactly that, yup. Sorry for the confusion :)
